etl/steps/data/garden/oecd/2024-04-10/income_distribution_database.py

```diff
@@ -0,0 +1,219 @@
+
+    tb = geo.harmonize_countries(
+        df=tb,
+        countries_file=paths.country_mapping_path,
+    )
+
+    tb = sanity_checks(tb)
+
+    tb = tb.format(["country", "year", "age"])
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def rename_and_create_columns(tb: Table) -> Table:
+    """
+    Rename categories in measure, poverty_line and age columns and make the table wide.
+    Also, add a gini_reduction column.
+    """
+    # Assert if all keys of dictionary are in the columns.
+    assert set(INDICATOR_NAMES.keys()) == set(
+        tb["measure"].unique()
+    ), "Not all expected categories are in the measure column"
+    assert set(POVERTY_LINES.keys()) == set(
+        tb["poverty_line"].unique()
+    ), "Not all expected categories are in the poverty_line column"
+    assert set(AGE_GROUPS.keys()) == set(tb["age"].unique()), "Not all expected categories are in the age column"
+
+    # Rename categories in measure, poverty_line and age columns.
+    tb["measure"] = tb["measure"].replace(INDICATOR_NAMES)
+    tb["poverty_line"] = tb["poverty_line"].replace(POVERTY_LINES)
+    tb["age"] = tb["age"].replace(AGE_GROUPS)
+
+    # Make the table wide, using measure as columns.
+    tb = tb.pivot(index=["country", "year", "poverty_line", "age"], columns="measure", values="value").reset_index()
+
+    # Create a variable that calculates the reduction from gini_market to gini_disposable
+    tb["gini_reduction"] = (tb["gini_market"] - tb["gini_disposable"]) / tb["gini_market"] * 100
+
+    return tb
+
+
+def create_relative_poverty_columns(tb: Table) -> Table:
+    """
+    Pivot table for headcount ratios and create multiple relative poverty columns with the poverty lines.
+    """
+
+    tb_inequality = tb.copy()
+    tb_poverty = tb.copy()
+
+    # Filter poverty_line column
+    tb_inequality = tb_inequality[tb_inequality["poverty_line"] == "Not applicable"].reset_index(drop=True)
+    tb_poverty = tb_poverty[tb_poverty["poverty_line"] != "Not applicable"].reset_index(drop=True)
+
+    # Define columns for both tables: tb_inequality has all the columns not containing headcount_ratio
+    # tb_poverty has all the columns containing headcount_ratio
+    inequality_columns = [c for c in tb_inequality.columns if "headcount_ratio" not in c]
+    poverty_columns = [c for c in tb_poverty.columns if "headcount_ratio" in c]
+
+    tb_inequality = tb_inequality[inequality_columns]
+    tb_poverty = tb_poverty[["country", "year", "poverty_line", "age"] + poverty_columns]
+
+    # Make tb_poverty wider
+    tb_poverty = tb_poverty.pivot(
+        index=["country", "year", "age"], columns="poverty_line", values=poverty_columns
```

Comment:
The `tables.pivot` function has a special argument that could be useful here, `join_column_levels_with="_"`. With that I think you would not need to reset indexes and rename columns (I haven't checked though).

Reply:
It works, thanks! I didn't know about it.